make more attributes optional

anvddriesch opened the pull request:
Towards https://github.com/giantswarm/giantswarm/issues/10786
Making the `InstanceType` in the `AWSControlPlane` optional, so that OPA can default it.
Making the  `Master` in the `AWSCluster` optional and adding a deprecation notice since we want to use the `AWSControlPlane` for the Master spec.

## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [x] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

anvddriesch commented:
I used `make` to generate the files but it seems like a lot of tests are failing now.

```

22:40 $ go test ./...
# github.com/giantswarm/apiextensions/github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:28:11: undefined: App
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:38:11: undefined: App
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:48:11: undefined: App
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:56:11: undefined: AppCatalog
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:65:11: undefined: AppCatalog
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:75:11: undefined: AppCatalog
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:83:11: undefined: AppCatalogList
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:98:11: undefined: AppCatalogList
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:108:11: undefined: AppCatalogList
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:116:11: undefined: AppCatalogSpec
github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1/zz_generated.deepcopy.go:116:11: too many errors
# github.com/giantswarm/apiextensions/github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:28:11: undefined: ETCDBackup
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:38:11: undefined: ETCDBackup
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:48:11: undefined: ETCDBackup
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:56:11: undefined: ETCDBackupList
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:71:11: undefined: ETCDBackupList
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:81:11: undefined: ETCDBackupList
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:89:11: undefined: ETCDBackupSpec
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:95:11: undefined: ETCDBackupSpec
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:105:11: undefined: ETCDBackupStatus
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:120:11: undefined: ETCDBackupStatus
github.com/giantswarm/apiextensions/pkg/apis/backup/v1alpha1/zz_generated.deepcopy.go:120:11: too many errors
# github.com/giantswarm/apiextensions/github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:28:11: undefined: MemcachedConfig
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:37:11: undefined: MemcachedConfig
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:41:13: undefined: MemcachedConfig
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:47:11: undefined: MemcachedConfig
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:55:11: undefined: MemcachedConfigList
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:61:17: undefined: MemcachedConfig
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:70:11: undefined: MemcachedConfigList
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:80:11: undefined: MemcachedConfigList
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:88:11: undefined: MemcachedConfigSpec
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:94:11: undefined: MemcachedConfigSpec
github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1/zz_generated.deepcopy.go:61:17: too many errors
# github.com/giantswarm/apiextensions/github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:28:11: undefined: AWSCluster
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:38:11: undefined: AWSCluster
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:48:11: undefined: AWSCluster
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:56:11: undefined: AWSClusterList
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:71:11: undefined: AWSClusterList
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:81:11: undefined: AWSClusterList
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:89:11: undefined: AWSClusterSpec
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:97:11: undefined: AWSClusterSpec
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:107:11: undefined: AWSClusterSpecCluster
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:116:11: undefined: AWSClusterSpecCluster
github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2/zz_generated.deepcopy.go:116:11: too many errors
# github.com/giantswarm/apiextensions/github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:28:11: undefined: AWSClusterConfig
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:37:11: undefined: AWSClusterConfig
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:47:11: undefined: AWSClusterConfig
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:55:11: undefined: AWSClusterConfigList
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:70:11: undefined: AWSClusterConfigList
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:80:11: undefined: AWSClusterConfigList
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:88:11: undefined: AWSClusterConfigSpec
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:96:11: undefined: AWSClusterConfigSpec
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:106:11: undefined: AWSClusterConfigSpecGuest
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:126:11: undefined: AWSClusterConfigSpecGuest
github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:126:11: too many errors
# github.com/giantswarm/apiextensions/github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:28:11: undefined: DeepCopyDate
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:35:11: undefined: DeepCopyDate
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:45:11: undefined: Release
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:54:11: undefined: Release
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:64:11: undefined: Release
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:72:11: undefined: ReleaseList
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:87:11: undefined: ReleaseList
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:97:11: undefined: ReleaseList
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:105:11: undefined: ReleaseSpec
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:125:11: undefined: ReleaseSpec
github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1/zz_generated.deepcopy.go:125:11: too many errors
# github.com/giantswarm/apiextensions/github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:28:11: undefined: AWSConfig
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:38:11: undefined: AWSConfig
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:48:11: undefined: AWSConfig
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:56:11: undefined: AWSConfigList
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:71:11: undefined: AWSConfigList
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:81:11: undefined: AWSConfigList
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:89:11: undefined: AWSConfigSpec
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:98:11: undefined: AWSConfigSpec
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:108:11: undefined: AWSConfigSpecAWS
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:130:11: undefined: AWSConfigSpecAWS
github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1/zz_generated.deepcopy.go:130:11: too many errors
# github.com/giantswarm/apiextensions/github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:28:11: undefined: AzureTool
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:38:11: undefined: AzureTool
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:48:11: undefined: AzureTool
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:56:11: undefined: AzureToolList
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:71:11: undefined: AzureToolList
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:81:11: undefined: AzureToolList
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:89:11: undefined: AzureToolSpec
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:96:11: undefined: AzureToolSpec
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:106:11: undefined: AzureToolStatus
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:112:11: undefined: AzureToolStatus
github.com/giantswarm/apiextensions/pkg/apis/tooling/v1alpha1/zz_generated.deepcopy.go:112:11: too many errors
ok      github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1       (cached)
ok      github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1      (cached)
ok      github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2    (cached)
ok      github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1  (cached)
ok      github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1   (cached)
ok      github.com/giantswarm/apiextensions/pkg/crd     (cached)
```
The same thing also happened to Marian in this branch so I think it's not because of my local setup. @tfussell do you maybe have an idea what is happening?

tfussell commented:
> The same thing also happened to Marian in this branch so I think it's not because of my local setup. @tfussell do you maybe have an idea what is happening?

I'm working on fixing a few issues with `apiextensions` that have come up since merging the PR. I'll be in touch to get you to test it soon.

marians commented on pkg/apis/infrastructure/v1alpha2/aws_cluster_types.go:
```diff
@@ -103,8 +103,10 @@ type AWSClusterSpecProvider struct {
+	// +kubebuilder:validation:Optional
 	// Master holds master node configuration details.
-	Master AWSClusterSpecProviderMaster `json:"master"`
+	// Note that this attribute is being deprecated. Master Spec can now be found in the AWSControlPlane
```
```suggestion
	// Note that this attribute is being deprecated. The master node specification can now be found in the AWSControlPlane resource.
```

anvddriesch commented:
By Thomas suggestion I deleted the `scripts` and `github` directories, moved the folder out of the gopath and hit `make` again. Tests are passing now. :)
